### ISSUE TITLE
A PyTorch version bump invalidates the execution floor for the training-speed comparison it exists to qualify

### ISSUE THREAD

**Issue by mmmfrieddough** [mmmfrieddough/anthro-chess#366]
## What was observed

`ENVIRONMENT_FIELDS` (`results/records.py:200`) is:

```python
("device", "device_name", "precision", "torch_version", "platform_key", "cpu_threads")
```

An execution floor is scoped to the hash of those fields, and `in_scope` requires **both** sides of a delta to carry the same scope:

```python
# results/noise.py:602
return measured == {scope}
```

So upgrading PyTorch changes the scope and the floor is refused — the delta reports `unknown`. That is precisely the comparison `anthro eval report --pivot environment` exists to make: *"pins the model and asks whether the machine, precision, or software did [change]."*

The asymmetry is the wrong way round for this project's actual use:

- **`torch_version` is in the key**, so a version bump blocks the comparison even though it is the same physical machine.
- **Driver and CUDA-runtime versions are not in the key at all**, so a driver change silently reuses a floor measured before it — the case where carrying the floor over is least defensible.

## Why this matters here

The metrics affected are the ones that answer whether a change made training or inference faster: eight `training.*` (`step_seconds`, `active_positions_per_second`, `peak_device_memory_bytes`, ...) and eight `inference.*` (latency p50/p90/p99, throughput, `model_load_seconds`). All are `measured_execution`.

Testing a PyTorch upgrade for a training-speed improvement is a comparison this project wants to make and currently cannot qualify.

## Scope

Small and deliberately bounded: let a delta that spans only a software-version change resolve a floor, rather than refusing it.

Decision 0025's rule is right about the case it was written for — a floor measured on a laptop must not qualify a workstation reading. The narrower question is which fields represent *the machine* and which represent *the thing under test*. A version bump on one host is the latter.

Where floors exist on both sides, combining them (the wider, or the root-sum-square) is well defined and both are already characterizable with `anthro eval noise sample` plus `characterize --kind execution`.

**Deliberately out of scope:** adding driver or CUDA-runtime versions to the environment record. Noted here as a known limitation rather than fixed — see the maintainer note below.

## Known limitation, accepted

A driver or CUDA-runtime upgrade does not invalidate an execution floor today and will not after this change. That is accepted rather than overlooked.

## References

- `docs/decisions/0025-machine-scoped-execution-noise-floors.md`
- `anthro eval report --pivot environment`

**Comment by mmmfrieddough** [comment]
Narrowed by `0043-a-delta-floor-is-combined-from-the-two-readings-it-compares.md`.

0043 removes the floor kinds for model quality but **keeps the execution floor and decision 0025's machine scoping for the efficiency families** — latency, throughput and training-step metrics vary with the machine rather than with the sample, and nothing in a reading's own units estimates that.

So this issue survives the collapse unchanged, and is now the *only* place floor scoping still exists. That makes it smaller and more clearly bounded than when filed: there is no longer a general scope system to keep consistent with, just the one rule for `training.*` and `inference.*`.

The known limitation recorded in the body — driver and CUDA-runtime versions are outside the environment key, so a driver change silently reuses a floor — is unaffected and still accepted.

**Comment by mmmfrieddough** [comment]
Expected to **dissolve** rather than be fixed, by #408.

This issue exists because an execution floor is stored apart from the reading and keyed to an environment hash that includes `torch_version`. #408 has the inference benchmark measure its own process dispersion during its run and report it beside the metric, like every other benchmark. With no stored floor and no environment key, comparing across a PyTorch upgrade just combines two dispersions.

Left open deliberately — it closes when #408 lands, not on the promise that it will.
